File: dl-invites/app/src/main/java/com/google/firebase/dynamicinvites/util/DynamicLinksUtil.java
package com.google.firebase.dynamicinvites.util;

import android.net.Uri;

import com.google.firebase.dynamicinvites.kotlin.model.InviteContent;
import com.google.firebase.dynamiclinks.DynamicLink;
import com.google.firebase.dynamiclinks.FirebaseDynamicLinks;

public class DynamicLinksUtil {

    public static InviteContent generateInviteContent() {
        return new InviteContent(
                "Hey check out my great app!",
                "It's like the best app ever.",
                generateContentLink());
    }

    // [START ddl_generate_content_link]
    public static Uri generateContentLink() {
        Uri baseUrl = Uri.parse("https://your-custom-name.page.link");
        String domain = "https://your-app.page.link";

        DynamicLink link = FirebaseDynamicLinks.getInstance()
                .createDynamicLink()
                .setLink(baseUrl)
                .setDomainUriPrefix(domain)
                .setIosParameters(new DynamicLink.IosParameters.Builder("com.your.bundleid").build())
                .setAndroidParameters(new DynamicLink.AndroidParameters.Builder("com.your.packageName").build())
                .buildDynamicLink();

        return link.getUri();
    }
    // [END ddl_generate_content_link]

}

File: dl-invites/app/src/main/java/com/google/firebase/dynamicinvites/kotlin/util/DynamicLinksUtil.kt
package com.google.firebase.dynamicinvites.kotlin.util

import android.net.Uri

import com.google.firebase.dynamicinvites.kotlin.model.InviteContent
import com.google.firebase.dynamiclinks.DynamicLink
import com.google.firebase.dynamiclinks.FirebaseDynamicLinks

object DynamicLinksUtil {

    fun generateInviteContent(): InviteContent {
        return InviteContent(
                "Hey check out my great app!",
                "It's like the best app ever.",
                generateContentLink())
    }

    // [START ddl_generate_content_link]
    fun generateContentLink(): Uri {
        val baseUrl = Uri.parse("https://your-custom-name.page.link")
        val domain = "https://your-app.page.link"

        val link = FirebaseDynamicLinks.getInstance()
                .createDynamicLink()
                .setLink(baseUrl)
                .setDomainUriPrefix(domain)
                .setIosParameters(DynamicLink.IosParameters.Builder("com.your.bundleid").build())
                .setAndroidParameters(DynamicLink.AndroidParameters.Builder("com.your.packageName").build())
                .buildDynamicLink()

        return link.uri
    }
    // [END ddl_generate_content_link]
}
